### PR TITLE
Pebble DB Integration into DB Bench Tool

### PR DESCRIPTION
**Pull request by praetoriansentry**
# Description

Geth has support for pebble db and it seems like bor will soon. This PR adds a mode flag to polycli to support testing a pebble db storage engine in addition to leveldb.

# Testing
```bash
# Create a baseline from leveldb testing
go run main.go dbbench --degree-of-parallelism 2 | jq '.' > result-level.json
# Move the database somewhere
mv _benchmark_db/ _benchmark_db.bak
# Run the same test but in pebble db mode
go run main.go dbbench --degree-of-parallelism 2 --db-mode pebbledb | jq '.' > result-pebble.json
```

Generally, it seems like pebble db is faster in all of the categories except for random reads. 
image

It does seem like there might be a pretty big difference in final state size
image

### CODE REVIEW

File: cmd/dbbench/dbbench.go [hunk 1]
```diff
@@ -372,42 +505,56 @@ benchLoop:
...
 			wg.Add(1)
 			go func() {
+				rCountLock.Lock()
 				rCount += 1
+				rCountLock.Unlock()
 				_ = pb.Add(1)
 
-				_, err := db.Get(rks.Key(), ro)
+				// It's not entirely obvious WHY this is needed, but without it, there are issues with the way that
+				// pebble db manages it's iterators and internal state. Level db works fine though.
```
Comment (praetoriansentry) [inline cmd/dbbench/dbbench.go:530]: @nivida fyi lol

File: cmd/dbbench/dbbench.go [hunk 2]
```diff
@@ -115,16 +118,156 @@ type (
...
+		NewIterator() iterator.Iterator
+		Get([]byte) ([]byte, error)
+		Put([]byte, []byte) error
+	}
+	LevelDBWrapper struct {
+		ro     *opt.ReadOptions
+		wo     *opt.WriteOptions
+		handle *leveldb.DB
+	}
+	PebbleDBWrapper struct {
```
Comment (leovct) [inline cmd/dbbench/dbbench.go:135]: nit: I would move most of the logic related to PebbleDB in `pebbledb.go` and same thing for LevelDB in `leveldb.go` just to keep it clean.

File: cmd/dbbench/usage.md [hunk 3]
```diff
@@ -2,7 +2,7 @@ This command is meant to give us a sense of the system level
 performance for leveldb:
 
 ```bash
-go run main.go leveldbbench --degree-of-parallelism 2 | jq '.' > result.json
+go run main.go dbbench --degree-of-parallelism 2 | jq '.' > result.json
```
Comment (leovct) [inline cmd/dbbench/usage.md:5]: nit: replace `go run main.go` with `polycli`

File: cmd/dbbench/usage.md [hunk 4]
```diff
@@ -2,7 +2,7 @@ This command is meant to give us a sense of the system level
...
 ```
 
 In many cases, we'll want to emulate the performance characteristics
```
Comment (leovct) [inline cmd/dbbench/usage.md:8]: unrelated to the PR, but curious to know how you get these values

**Review by leovct — APPROVED** [review]
lgtm!

File: cmd/dbbench/dbbench.go (hunk 2 above)
Comment (praetoriansentry) [reply cmd/dbbench/dbbench.go:135]: I can move the pebbledb and leveldb implementations. I think that the common interferface between the two makes sense in the main package with the command.  RandomKeySeeker, I have no point of view, it's just a utility class that I needed for now I'll probably leave it where it is

**Pull request merged**